doc/BorderRouter.md

```diff
@@ -0,0 +1,124 @@
+updates the pointers of the ipv4.Message.Buffers to the buffer inside those packets and then performs a batch read.
+* **Processing Routines** There are several processing routines and slow-path processing routines
+in the border router that are responsible for processing the received packet.
+The actual processing logic remains unchanged.
+If the processing routine identifies a packet that belongs to the slow-path, the processing routines enqueues
+the packet to a slow-path processing routine. If the queue of the slow-path processing routine is full, the
+packet will not be processed at all. In this case the buffer is immediately returned to the receiver.
+Once a packet is processed, it gets enqueued to the forwarder which is responsible for the egress interface.
+If the queue of the forwarder is full, the packet will be dropped.
+* **Forwarders** There exists one forwarder per border router interface that is responsible for forwarding the
+packets over the network that it receives from the processing routines. It forwards the packets as a batch.
+Afterwards it returns each buffer to the receiver from which it originates.
+
+![Border Router Design](fig/border_router/br_design.png)
+
+## Mapping of processing routines
+To prevent any packet reordering on the fast-path, we map the tuple of source and flowID, see the
+[SCION header documentation](https://github.com/scionproto/scion/blob/master/doc/protocols/scion-header.rst),
+to a fixed processing routine using a hash function together with a secret value to prevent pre-computations
+of the exact mapping.
+
+## Slow path
+During processing, packets that have to follow the slow path are identified and forwarded to the
+slow-path processing routines.
+Rate limiting of slow-path operations is not implemented explicitly, but only implictily through specifying
+the number of slow-path processing routines in the configuration.
+In case a packet is identified to belong to the slow path but the queue of the slow path is full, the
+packet is dropped.
+Packets currently identified for slow-path are:
+* Malformed packets
+* SCMP traceroute packets
+
+# Configuration
+The configuration of the border router will remain in the border router toml file.
+The following configuration entries are added:
+
+## Pool size
+Since a pool of packet buffers is bound to a receiver, the size of the pool can be configured for each
+receiver seperately.
+It makes sense to make this configurable on a per border router interface level because not every 
+interface might have the same expected load and hence not every interface requires the same pool size.
+An appropriate default value would have to be evaluated.
+
+## Number of processing routines (N)
+By configuring the number of processing routines one can specify the number of goroutines that are able
+to process packets in parallel.
+A default value could be the number of border router interfaces.
+
+## Number of slow-path processing routines (M)
```

Comment:
I think a percentage of the number of processing routines fit better than a percentage of received packets. Otherwise we would have to monitor how many we receive and update the number of slow-path processing routines while the border router is running which would not be trivial.